source/docs/advanced/graphql/dataloaders-and-cache-invalidation.md

```diff
@@ -0,0 +1,396 @@
+
+The dataLoader cache must be configured in the [`config/caching.js` configuration file](/docs/reference/configurations.html#config-caching-js). Please refer to the reference documentation for further details or read the following section to choose the most relevant strategies for your context.
+
+## Caching strategies
+
+This section provides details about the available strategies implementations in Front-Commerce. They can be used by using them in the `implementation` key of your caching strategies configuration.
+
+### `Redis`
+
+The `Redis` implementation allows to cache data in a redis server.
+
+It uses node's [`redis` package](https://www.npmjs.com/package/redis) as a redis client. Please refer to the package documentation for [all the available options](https://www.npmjs.com/package/redis#options-object-properties).
+
+Here is a configuration example:
+
+```js
+// my-module/config/caching.js
+export default {
+  strategies: [
+    {
+      implementation: "Redis",
+      supports: "*",
+      config: {
+        // see https://www.npmjs.com/package/redis#options-object-properties
+        host: "127.0.0.1",
+        port: 6379,
+        db: 1
+      }
+    }
+  ]
+};
+```
+
+### `PerMagentoCustomerGroup`
+
+<blockquote class="feature--new">
+_Since version 2.0.0-rc.0_
+</blockquote>
+
+The `PerMagentoCustomerGroup` implementation is a decorator that is specific to Magento 1 and Magento 2 modules. It will decorate the existing caching strategies so that dataLoader keys are specific to the current customer group. We highly recommend to use it on Magento stores that have price per group, so they can leverage other caching mechanisms (such as `Redis`).
+
+It is possible to provide a default group to use as scope for guest users.
+
+Here is a configuration example:
+
+```js
+// my-module/config/caching.js
+export default {
+  strategies: [
+    // The PerMagentoCustomerGroup strategy MUST be registered after a persistent cache implementation
+    // because it has no effect in the context of the default per-request in-memory caching.
+    {
+      implementation: "Redis",
+      supports: "*",
+      config: {
+        host: "127.0.0.1"
+      }
+    },
+    {
+      implementation: "PerMagentoCustomerGroup",
+      // Will scope all data from the CatalogPrice dataLoader with the customer group
+      // before they are transmitted to the previous strategy (Redis).
+      // Other dataLoaders will use Redis storage in a standard fashion.
+      supports: ["CatalogPrice"],
+      config: {
+        defaultGroupId: 0
+      }
+    }
+  ]
+};
+```
+
+### Advanced usage
+
+If you need additional implementations or want to leverage strategies for a specific use case, please [contact us](mailto:contact@front-commerce.com) so we can discuss it and guide you!
+
+## Invalidating the cache
```

Comment:
Not sure when it'll make sense to document it, but it might prove to be useful to expose/document the node API to invalidate cache directly. I'm thinking of systems that only have webhooks (SaaS platforms mainly) where we can add a server endpoint that will then call the correct invalidation method in node.

Reply:
I suggest that we wait for a few implementations of this pattern to go live before documenting it. I feel like some abstraction may emerge.